lib/std/io/early_eof_reader.zig

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const std = @import("../std.zig");
+const io = std.io;
+const assert = std.debug.assert;
+const testing = std.testing;
+
+pub fn EarlyEOFReader(comptime ReaderType: type) type {
```

Comment:
Oh `EarlyEOFReader` may be really useful, I like it.
The name doesn't really convey what it does though, I'd go with `LimitedReader` (name borrowed from [golang](https://golang.org/pkg/io/#LimitedReader) or something along that line.

Reply:
hrm. I agree the name isn't great, but I don't think that `LimitedReader` is better...

Open to other suggestions.

Reply:
Maybe some variant of `MaxLengthReader`, `FixedSizeReader`, or something idk.. 

EarlyEOFReader was really confusing before reading what it does.

Reply:
Or `CappedReader`